src/main/java/at/hannibal2/skyhanni/data/mob/MobDetection.kt

```diff
@@ -354,4 +344,21 @@ class MobDetection {
         }
     }
 
+    @SubscribeEvent
+    fun onConfigLoad(event: ConfigLoadEvent) {
+        ConditionalUtils.onToggle(config.enabled) {
+            clearMobs()
+        }
+    }
+
+    private fun clearMobs() {
+        mobDetectionReset()
+        MobData.previousEntityLiving.clear()
+        MobData.currentEntityLiving.clear()
+        MobData.retries.clear()
+        MobData.entityToMob.clear()
```

Comment:
this breaks the contract.
Better way is to check if it is empty else show a error with the data.